modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/storage/LastSyncGlobalSnapshotStorage.scala

```diff
@@ -99,32 +102,44 @@ object LastSyncGlobalSnapshotStorage {
       ): F[Option[List[Hashed[GlobalIncrementalSnapshot]]]] =
         currencySnapshotStorage.head.flatMap {
           case Some((_, info)) =>
-            info.globalSnapshotSyncView.flatTraverse { syncView =>
-              val lastOrdinalOpt = syncView.values
-                .map(_.globalSnapshotOrdinal)
-                .groupBy(identity)
-                .toList
-                .sortBy { case (ordinal, occurrences) => (-occurrences.size, ordinal.value.value) }
-                .map { case (ordinal, _) => ordinal }
-                .maxOption
-
-              lastOrdinalOpt match {
-                case Some(lastOrdinal) =>
-                  val selectedOrdinals = (0 until n).flatMap { i =>
-                    NonNegLong.from(lastOrdinal.value.value - i.toLong).toOption.flatMap(SnapshotOrdinal(_))
-                  }.toSet.toList
-
-                  selectedOrdinals.traverse(get).map { results =>
-                    val snapshots = results.flatten
-                    if (snapshots.isEmpty) None else Some(snapshots)
-                  }
-
-                case None =>
-                  Async[F].pure(None)
+            val maybeSnapshots: F[Option[List[Hashed[GlobalIncrementalSnapshot]]]] =
+              info.globalSnapshotSyncView.fold(
+                Async[F].pure(Option.empty[List[Hashed[GlobalIncrementalSnapshot]]])
+              ) { syncView =>
+                val mostFrequentOrdinal = syncView.values
+                  .map(_.globalSnapshotOrdinal)
+                  .groupBy(identity)
+                  .toList
+                  .sortBy { case (ordinal, occurrences) => (-occurrences.size, ordinal.value.value) }
+                  .map { case (ordinal, _) => ordinal }
+                  .headOption
```

Comment:
Could this lead to a fork if we are using the most frequency global ordinal instead of requiring a majority? Though perhaps I am misunderstanding the wider context of when this is called.